iotdevice/transport/http/http.go

```diff
@@ -138,41 +164,43 @@ func (tr *Transport) GetBlobSharedAccessSignature(ctx context.Context, blobName
 		return "", "", err
 	}
 
-	resourceURI := fmt.Sprintf("%s/%s", tr.creds.GetHostName(), tr.creds.GetDeviceID())
-	sas, err := tr.creds.Token(resourceURI, tr.ttl)
+	response, err := tr.getTokenAndSendRequest(http.MethodPost, target, requestPayloadBytes)
 	if err != nil {
 		return "", "", err
 	}
 
-	request, err := http.NewRequest(http.MethodPost, target.String(), bytes.NewReader(requestPayloadBytes))
+	err = tr.handleErrorResponse(response)
 	if err != nil {
 		return "", "", err
 	}
-	request.Header.Add("Content-Type", "application/json; charset=utf-8")
-	request.Header.Add("Authorization", sas.String())
-
-	response, err := tr.client.Do(request)
+	var responsePayload BlobSharedAccessSignatureResponse
+	err = json.NewDecoder(response.Body).Decode(&responsePayload)
 	if err != nil {
 		return "", "", err
 	}
 
-	if response.StatusCode != http.StatusOK {
-		var responsePayload ErrorResponse
-		err = json.NewDecoder(response.Body).Decode(&response)
-		if err != nil {
-			return "", "", err
-		}
+	return responsePayload.CorrelationID, responsePayload.SASURI(), nil
+}
 
-		return "", "", fmt.Errorf("code = %d, message = %s, exception message = %s", response.StatusCode, responsePayload.Message, responsePayload.ExceptionMessage)
+func (tr *Transport) getTokenAndSendRequest(method string, target *url.URL, requestPayloadBytes []byte) (*http.Response, error) {
+	resourceURI := fmt.Sprintf("%s/%s", tr.creds.GetHostName(), tr.creds.GetDeviceID())
+	sas, err := tr.creds.Token(resourceURI, tr.ttl)
+	if err != nil {
+		return nil, err
 	}
 
-	var responsePayload BlobSharedAccessSignatureResponse
-	err = json.NewDecoder(response.Body).Decode(&responsePayload)
+	request, err := http.NewRequest(method, target.String(), bytes.NewReader(requestPayloadBytes))
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
+	request.Header.Add("Content-Type", "application/json; charset=utf-8")
+	request.Header.Add("Authorization", sas.String())
 
```

Comment:
To update an existing module we need to use `If-Match` in the header, either we need to pass `eTag` of the module or `*` to accept all incoming for the update
`request.Header.Add("If-Match", "*")`

Reply:
So I added the etag option to the model and added the if-match header to the request